### PR TITLE
CPS-0014? | Register of CBOR Tags

### PR DESCRIPTION
**Pull request by stevenj**
This is the first of three PRs that should be considered together for context.

See also:

* #752 
* #753 

Closes input-output-hk/catalyst-CIPs#2

---

([draft rendered from working branch](https://github.com/input-output-hk/catalyst-CIPs/blob/cps/cbor-tags/CPS-0014/README.md))

### CODE REVIEW

**Review by Ryun1 — COMMENTED** [review]
Very interesting problem statement, I believe this to be a genuine problem.
Im keen to hear other opinions on it.

File: CPS-xxxx/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+An example is `ULID` this has no current CBOR Tag, but it is not defined by a CIP nor integral to Cardano.
+Even though a CIPs metadata may use one, its formalization should be done outside the CIP process.
+
+## Open Questions
+
+* Should we define a canonical list of tags, the CIP they are defined in, and if they are IANA registered or not?
+* Who should register the Tag with IANA?
+  * Would this become a necessary part of the "path to Active" of any CIP which defines a CBOR tag?
+* How should duplicate structures be handled (same data encoded differently with a different tag)?
+* Should defining tags be their own CIP, or can they be part of a larger CIP?
```
[the remark this answers is not in the document]
Comment (stevenj) [reply CPS-xxxx/README.md:110]: I added a potential registry to the draft CIP solution to this CPS

**Review by Crypto2099 — APPROVED** [review]
This is well written and looks largely complete from a CPS perspective. Happy to promote this one for consideration at the next meeting and see if we can get it numbered up.

File: CPS-xxxx/LICENSE [hunk 2]
```diff
@@ -0,0 +1,395 @@
+Attribution 4.0 International
```
Comment (Ryun1) [inline CPS-xxxx/LICENSE:1]: @rphair 

with authors adding a license like this, perhaps it does make sense to add a copyright section to CPS

File: CPS-xxxx/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Status: Open
+Category: Meta
+Authors:
+  - Steven Johnson <steven.johnson@iohk.io>
+Proposed Solutions:
+  - https://github.com/cardano-foundation/CIPs/pull/752
+Discussions:
+  - https://github.com/cardano-foundation/CIPs/pull/751
+Created: 2024-01-24
+License: CC-BY-4.0
```
Comment (Ryun1) [inline CPS-xxxx/README.md:13]:
```suggestion
```
Please remove!
adding to the header will break downstream tooling

**Review by rphair — COMMENTED** [review]
Well received along with 2 related CIPs; all promoted to candidates at CIP meeting today. 🎉

File: CPS-xxxx/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
+---
+CPS: xxxx
```
Comment (rphair) [inline CPS-xxxx/README.md:2]:
```suggestion
CPS: 14
```
Also please rename the document link in the OP.

**Review by rphair — COMMENTED** [review]
p.s. - the elements of my https://github.com/cardano-foundation/CIPs/pull/751#pullrequestreview-1866039329 whose code comments were eaten by GitHub in some unknowable way...

Firstly please delete the `LICENSE` file from your branch.  We're adding licensing to all CPSs with a framework forthcoming as per https://github.com/cardano-foundation/CIPs/issues/759 and before merge we'll probably add a copyright field to the YAML header & Markdown footer.

**Comment by rphair** [comment]
@stevenj we've also had consensus at the CIP meeting today that these CBOR subssions (this and two subsidiary CIPs) should have a category `Tools` because of their potential to produce & consume these forms across a wide variety of off-chain contexts.

File: CPS-xxxx/LICENSE (hunk 2 above)
Comment (rphair) [reply CPS-xxxx/LICENSE:1]: yes @Ryun1 as of https://github.com/cardano-foundation/CIPs/pull/762#issuecomment-1934233812... @stevenj please add CPS style licensing as you would for a CIP, in both the header and footer.

File: CPS-xxxx/README.md (hunk 3 above)
Comment (rphair) [reply CPS-xxxx/README.md:13]: yes @stevenj please do (see https://github.com/cardano-foundation/CIPs/pull/751#discussion_r1483370143)

File: CPS-xxxx/README.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Title: Register of CBOR Tags for Cardano Data structures
+Status: Open
+Category: Meta
```
Comment (rphair) [inline CPS-xxxx/README.md:5]:
```suggestion
Category: Tools
```

**Review by rphair — COMMENTED** [review]
continuing https://github.com/cardano-foundation/CIPs/pull/751#issuecomment-1930547443:

File: CPS-xxxx/README.md (hunk 3 above)
Comment (stevenj) [reply CPS-xxxx/README.md]: Updated as requested

**Comment by stevenj** [comment]
@rphair @Ryun1 This CPS is now numbered properly, and the license notifications have been updated as requested.

File: CPS-0014/README.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+CPS: 14
+Title: Register of CBOR Tags for Cardano Data structures
+Status: Open
+Category: Tools
+Authors:
+  - Steven Johnson <steven.johnson@iohk.io>
+Proposed Solutions:
+  - https://github.com/cardano-foundation/CIPs/pull/752
+Discussions:
+  - https://github.com/cardano-foundation/CIPs/pull/751
```
Comment (Ryun1) [inline CPS-0014/README.md:11]:
```suggestion
  - https://github.com/cardano-foundation/CIPs/pull/751
  - https://github.com/cardano-foundation/CIPs/pull/752
```
Add the CIP in

File: CPS-0014/README.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
+---
+CPS: 14
+Title: Register of CBOR Tags for Cardano Data structures
+Status: Open
+Category: Tools
+Authors:
+  - Steven Johnson <steven.johnson@iohk.io>
+Proposed Solutions:
+  - https://github.com/cardano-foundation/CIPs/pull/752
```
Comment (Ryun1) [inline CPS-0014/README.md:9]:
```suggestion
  - https://github.com/cardano-foundation/CIPs/tree/master/CIP-0114
```
Preempt the merge

File: CPS-0014/README.md (hunk 6 above)
Comment (stevenj) [reply CPS-0014/README.md:11]: Done

**Review by Ryun1 — APPROVED** [review]
LGTM !

**Review by rphair — APPROVED** [review]
All review details apparently addressed as for the related CIPs. 🚀

**Pull request merged**